Exo trigger prompt monitoring92 x

gomber opened the pull request.

gomber committed:
Added prompt monitoring code for HLT Singlephoton paths

gomber committed:
Added 2016 HLT pahts for jets and photon to make it run

gomber committed:
Added 2016 HLT pahts for jets and photon to make it run

gomber committed:
Added 2016 HLT pahts for jets and photon to make it run

gomber committed:
Added displaced photon paths

gomber committed:
Added displaced photon paths

gomber committed:
Added more plots, efficiency as a fucntion of phootn and eta

cmsbuild commented:
A new Pull Request was created by  @gomber  for master.

It involves the following packages:

Calibration/HcalAlCaRecoProducers
DQMOffline/Trigger


@ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @tocheng, @mmusich, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mtosi commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,302 @@
+  histname = "photon_eta"; histtitle = "Photon eta";
+  bookME(ibooker,photonEtaME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+  setTitle(photonEtaME_,"Photon #eta","events / 0.1");
+
```
aren't you missing the 2D plot ?

gomber replied:
which 2D plots you want? i am just making trigger efficiency plot as a function of pt, eta and phi, so that should be enough

mtosi replied:
in addition, are you planning to add variables for the "out-of-time" photon ?

gomber replied:
I mean having eta and phi 1D plot is sufficient. And yes we would like to add out-of-time photons variable for displaced photons , kevin can comment?

mtosi commented on DQMOffline/Trigger/python/PhotonMonitor_cff.py:
```diff
@@ -0,0 +1,47 @@
+#HLT_SinglePhoton200_IDTight
+SinglePhoton300_monitoring = hltPhotonmonitoring.clone()
+SinglePhoton300_monitoring.FolderName = cms.string('HLT/Photon/Photon300/')
+SinglePhoton300_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Photon300_NoHE_v")
```
aren't you apply any offline selection ?

mtosi commented on DQMOffline/Trigger/python/PhotonMonitor_cff.py:
```diff
@@ -0,0 +1,47 @@
+# HLT_SinglePhoton175_IDTight
+SinglePhoton175_monitoring = hltPhotonmonitoring.clone()
+SinglePhoton175_monitoring.FolderName = cms.string('HLT/Photon/Photon175/')
+SinglePhoton175_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Photon175_v*")
```
aren't you apply any offline selection ?

gomber replied:
there is photonselection in the python file photonMonitoring_cfi, which will be created from .cc file.
 desc.add<std::string>("photonSelection", "pt > 145 && eta<1.4442 && hadTowOverEm<0.0597 && full5x5_sigmaIetaIeta()<0.01031 && chargedHadronIso<1.295");

mtosi commented on DQMOffline/Trigger/python/PhotonMonitor_cff.py:
```diff
@@ -0,0 +1,47 @@
+Photon60_monitoring = hltPhotonmonitoring.clone()
+Photon60_monitoring.FolderName = cms.string('HLT/Photon/Photon60/')
+hltPhotonmonitoring.denGenericTriggerEventPSet.hltPaths = cms.vstring()
+hltPhotonmonitoring.denGenericTriggerEventPSet.andOrHlt_ = cms.bool(True)
```
I'm confused :(
are you using any HLT paths for the denominator ?
why are you setting
```
hltPhotonmonitoring.denGenericTriggerEventPSet.hltPaths = cms.vstring()
hltPhotonmonitoring.denGenericTriggerEventPSet.andOrHlt_ = cms.bool(True)

```?

mtosi commented:
efficiency in the eta-phi plane ;)
  mia

On Fri, May 26, 2017 at 8:37 PM, gomber <notifications@github.com> wrote:

> *@gomber* commented on this pull request.
> ------------------------------
>
> In DQMOffline/Trigger/plugins/PhotonMonitor.cc
> <https://github.com/cms-sw/cmssw/pull/18971#discussion_r118765312>:
>
> > +  bookME(ibooker,photonME_variableBinning_,histname,histtitle,photon_variable_binning_);
> +  setTitle(photonME_variableBinning_,"Photon pT [GeV]","events / [GeV]");
> +
> +  histname = "photonVsLS"; histtitle = "photon pt vs LS";
> +  bookME(ibooker,photonVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,photon_binning_.xmin, photon_binning_.xmax);
> +  setTitle(photonVsLS_,"LS","Photon pT [GeV]");
> +
> +  histname = "photon_phi"; histtitle = "Photon phi";
> +  bookME(ibooker,photonPhiME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
> +  setTitle(photonPhiME_,"Photon #phi","events / 0.1 rad");
> +
> +
> +  histname = "photon_eta"; histtitle = "Photon eta";
> +  bookME(ibooker,photonEtaME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
> +  setTitle(photonEtaME_,"Photon #eta","events / 0.1");
> +
>
> which 2D plots you want? i am just making trigger efficiency plot as a
> function of pt, eta and phi, so that should be enough
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18971#discussion_r118765312>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58wA29CzAZl9O6anrutNFRI-5yEdFks5r9xvrgaJpZM4Nn66Y>
> .
>

arunhep commented:
i think this PR needs rebase.

gomber committed:
small changes

cmsbuild commented:
Pull request #18971 was updated. @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #18971 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
@arunhep should be fine now..

gomber commented:
So guys, I am adding 2D plot and will do another commit as Mia suggested and then we would be good to go!

mtosi commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,302 @@
+
+
+  histname = "photon_eta"; histtitle = "Photon eta";
+  bookME(ibooker,photonEtaME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
```
shouldn't be eta_binning ?

gomber replied:
phi_binning which we are using in the code is very general, and can be used for eta too, its 64 bins between -3.2 to 3.2,  thats why used the same variable

gomber replied:
Ok I am using the following one now, since we are mainly interested in barrel region for these, so looking at the ECAL tower size, 1.4442/0.087 =  ~17 bins, so I will cover -1.4442 to 1.4442 with 34 bins.

mtosi commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,302 @@
+  fillHistoPSetDescription(metPSet);
+  histoPSet.add<edm::ParameterSetDescription>("photonPSet", metPSet);
+  std::vector<double> bins = {0.,20.,40.,60.,80.,90.,100.,110.,120.,130.,140.,150.,160.,170.,180.,190.,200.,220.,240.,260.,280.,300.,350.,400.,450.,1000.};
+  histoPSet.add<std::vector<double> >("photonBinning", bins);
```
I think you should overwrite this in the python config
taking into account the expected turnON

mtosi commented:
please use the more appropriate for photon obj

thanks
  mia

On Fri, May 26, 2017 at 9:52 PM, gomber <notifications@github.com> wrote:

> *@gomber* commented on this pull request.
> ------------------------------
>
> In DQMOffline/Trigger/plugins/PhotonMonitor.cc
> <https://github.com/cms-sw/cmssw/pull/18971#discussion_r118778006>:
>
> > +
> +  histname = "photon_pt_variable"; histtitle = "photon PT";
> +  bookME(ibooker,photonME_variableBinning_,histname,histtitle,photon_variable_binning_);
> +  setTitle(photonME_variableBinning_,"Photon pT [GeV]","events / [GeV]");
> +
> +  histname = "photonVsLS"; histtitle = "photon pt vs LS";
> +  bookME(ibooker,photonVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,photon_binning_.xmin, photon_binning_.xmax);
> +  setTitle(photonVsLS_,"LS","Photon pT [GeV]");
> +
> +  histname = "photon_phi"; histtitle = "Photon phi";
> +  bookME(ibooker,photonPhiME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
> +  setTitle(photonPhiME_,"Photon #phi","events / 0.1 rad");
> +
> +
> +  histname = "photon_eta"; histtitle = "Photon eta";
> +  bookME(ibooker,photonEtaME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
>
> phi_binning which we are using in the code is very general, and can be
> used for eta too, its 64 bins between -3.2 to 3.2, thats why used the same
> variable
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18971#discussion_r118778006>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt582Eadk-RrjLO2v4U4KsIw5-ym2VRks5r9y17gaJpZM4Nn66Y>
> .
>

mtosi commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,302 @@
+  photonME_variableBinning_.denominator = nullptr;
+  photonVsLS_.numerator   = nullptr;
+  photonVsLS_.denominator = nullptr;
+  photonEtaME_.numerator   = nullptr;
```
I think the variable binning should be used for eta as well

mtosi commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,302 @@
+  photonEtaME_.denominator = nullptr;
+  photonPhiME_.numerator   = nullptr;
+  photonPhiME_.denominator = nullptr;
+
```
I would suggest to add the 2D plot

mtosi commented on DQMOffline/Trigger/python/PhotonMonitor_cff.py:
```diff
@@ -0,0 +1,46 @@
+Photon60_monitoring = hltPhotonmonitoring.clone()
+Photon60_monitoring.FolderName = cms.string('HLT/Photon/Photon60/')
+Photon60_monitoring.denGenericTriggerEventPSet.hltPaths = cms.vstring()
+Photon60_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Photon60_R9Id90_CaloIdL_IsoL_v*")
```
are you sure it is available in 2017 ?

gomber replied:
yes it is available kevin requested them
https://its.cern.ch/jira/browse/CMSHLT-1327

mtosi commented on DQMOffline/Trigger/python/PhotonMonitor_cff.py:
```diff
@@ -0,0 +1,46 @@
+
+Photon60_DisplacedIdL_PFJet350MinPFJet15_monitoring = Photon60_DisplacedIdL_monitoring.clone()
+Photon60_DisplacedIdL_PFJet350MinPFJet15_monitoring.FolderName = cms.string('HLT/Photon/Photon60_DisplacedIdL_PFJet350MinPFJet15/')
+Photon60_DisplacedIdL_PFJet350MinPFJet15_monitoring.denGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Photon60_R9Id90_CaloIdL_IsoL_v*","HLT_PFHT350MinPFJet15_v*")
```
are you sure they are available ?

gomber replied:
see previous message

mtosi commented:
what about
HLT_Photon50_R9Id90_HE10_IsoM_v8
HLT_Photon75_R9Id90_HE10_IsoM_v8
HLT_Photon90_R9Id90_HE10_IsoM_v8
HLT_Photon120_R9Id90_HE10_IsoM_v8
HLT_Photon165_R9Id90_HE10_IsoM_v9

gomber commented:
Hi Mia, 

who requested these paths, I didn't see it in Sam mails, may be I missed it?
And I thought the idea of prompt monitoring is for signal paths, so that we can find the issues (if any) fast. Another question, i assume these paths are prescale?

Regards
Bhawna

fwyzard commented:
We need monitoring for all triggers.

gomber committed:
Fixed denominator for HLTPhoton Displaced paths

cmsbuild commented:
Pull request #18971 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

gomber committed:
Added 2d eta-phi, r9, hoverE plots and added other photons paths

cmsbuild commented:
Pull request #18971 was updated. @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
Hi Mia, All

I added all the relevant plots as you requested  (2d plots, H/E and r9) and also added other photon paths.
HLT_Photon50_R9Id90_HE10_IsoM_v8
HLT_Photon75_R9Id90_HE10_IsoM_v8
HLT_Photon90_R9Id90_HE10_IsoM_v8
HLT_Photon120_R9Id90_HE10_IsoM_v8
HLT_Photon165_R9Id90_HE10_IsoM_v9

I need to see how to refer to the below PR so that DQMGeneric client producer part i is taken care of :https://github.com/cms-sw/cmssw/pull/18751, but rest is all done.

Regards
Bhawna

mtosi commented:
great!
i'll give a look at the update as soon as i'll be in the office

for the client, as far as i understood,
it is probably just a change 
cms.EDAnalyzer
-->
cms.EDProducer

@dmitrijus, could you please comment on this ?
thanks !!!

gomber committed:
Changed cms.EdAnalyzer-> DQMEDHarvester

cmsbuild commented:
Pull request #18971 was updated. @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented on Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalMinBiasHI_cff.py:
```diff
@@ -79,7 +79,8 @@
 from Configuration.Eras.Modifier_run2_HCAL_2017_cff import run2_HCAL_2017
 run2_HCAL_2017.toModify( hbherecoNoise,
     processQIE11 = cms.bool(True),
-    setNoiseFlagsQIE11 = cms.bool(True),
```
I think you need to rebase

cmsbuild commented:
Pull request #18971 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

dmitrijus commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,351 @@
+  me.numerator   = ibooker.bookProfile(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, xmin, xmax, ymin, ymax);
+  me.denominator = ibooker.bookProfile(histname+"_denominator", histtitle+" (denominator)", nbinsX, xmin, xmax, ymin, ymax);
+}
+void PhotonMonitor::bookME(DQMStore::IBooker &ibooker, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, int nbinsY, double ymin, double ymax)
```
These method overloads are very hard to follow, why not just name them for what they are supposed to do?
bookME2D or book2DME or bookProfileME

The main worry of mine is the added memory consumption.

dmitrijus commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

mtosi commented:
they come from my original example/template, I'm afraid
do they really bring memory consumption named like that ?

dmitrijus commented:
No. Naming like that just makes it harder to see how many bins are they using.
I will wait for the tests and see it the empirical change, if any :)

cmsbuild commented:
-1

Tested at: 69dbfa5537432af7c8f9cc06c1086a0229076462

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20253/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/src/EgHLTOffEvt.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/src/HLTTauDQMPathPlotter.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/src/JetMETHLTOfflineSource.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/src/EventShapeDQM.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/src/EgHLTMonElemFuncs.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:35:5: error: redefinition of 'int N_r9'
 int N_r9 = 50;
     ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:26:5: note: 'int N_r9' previously defined here
 int N_r9 = 50;
     ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/src/SealModule.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/BPHMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/METMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/DQMGenericTnPClient.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:35:5: error: redefinition of 'N_r9'
int N_r9 = 50;
    ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-31-1100/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:26:5: note: previous definition is here
int N_r9 = 50;
    ^
</pre>

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

gomber committed:
Removed duplicate definition of N_r9

cmsbuild commented:
Pull request #18971 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
I tested it again and it worked fine in my area with the little change i committed earlier, so please go ahead and test it.
Sorry about that!

Regards
Bhawna

mtosi commented:
@dmitrijus could we start the integration test ?

gomber commented:
Hi Guys?

@dmitrijus Could you please do the integration test, so that we can finalize this PR?

Regards
Bhawna

dmitrijus commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,350 @@
+{
+  int nbins = binning.size()-1;
+  std::vector<float> fbinning(binning.begin(),binning.end());
+  float* arr = &fbinning[0];
```
use fbinning.data() instead.

davidlange6 replied:
it looks like this comment is still to be addressed?

dmitrijus commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2870212bf188aa760da6c119114382f8a47f9ae4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20373/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20373/summary.html

Comparison Summary:
 - You potentially added 40 lines to the logs
 - Reco comparison results: 1697 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1837639
 - DQMHistoTests: Total failures: 38362
 - DQMHistoTests: Total nulls: 65
 - DQMHistoTests: Total successes: 1799039
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

fwyzard commented:
tracked by #19142

mtosi commented:
can we proceed w/ this ?

mtosi commented:
@dmitrijus, if it is fine w/ you, can you please sign it ?
thanks !

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,350 @@
+
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+double MAX_PHI1 = 3.2;
```
hi @gomber 
no global variables - just move these into the scope of the Photon Monitor class

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,350 @@
+  if (den_genTriggerEventFlag_) delete den_genTriggerEventFlag_;
+}
+
+MEbinning PhotonMonitor::getHistoPSet(edm::ParameterSet pset)
```
pass pset as a const &

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,350 @@
+      };
+}
+
+void PhotonMonitor::setTitle(PhotonME& me, std::string titleX, std::string titleY)
```
pass the strings as const &

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,350 @@
+  iEvent.getByToken( jetToken_, jetHandle );
+  std::vector<reco::PFJet> jets;
+  jets.clear();
+  if ( int(jetHandle->size()) < njets_ ) return;
```
make njets_ a unsigned int to avoid the need for int()

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,350 @@
+  for ( auto const & e : *eleHandle ) {
+    if ( eleSelection_( e ) ) electrons.push_back(e);
+  }
+  if ( int(electrons.size()) < nelectrons_ ) return;
```
and similarly here.

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -0,0 +1,350 @@
+  if (num_genTriggerEventFlag_->on() && ! num_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  // filling histograms (num_genTriggerEventFlag_)  
+  if(photons.size()>0)
```
use photons.empty()

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.h:
```diff
@@ -0,0 +1,121 @@
+
+
+
+  GenericTriggerEventFlag* num_genTriggerEventFlag_;
```
please make these unique_ptrs

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.h:
```diff
@@ -0,0 +1,121 @@
+
+#include <string>
+#include <vector>
+#include <map>
```
it looks like this include is not needed

davidlange6 commented on DQMOffline/Trigger/plugins/PhotonMonitor.h:
```diff
@@ -0,0 +1,121 @@
+  GenericTriggerEventFlag* num_genTriggerEventFlag_;
+  GenericTriggerEventFlag* den_genTriggerEventFlag_;
+
+  StringCutObjectSelector<reco::MET,true>         metSelection_;
```
do you expect significantly more complex selections than those in the fillDescriptions? If not, the StringCutObjectSelector is very slow and memory intensive for your needs

davidlange6 replied:
ah - never mind - I see a complex one now

gomber committed:
changing fbinning

cmsbuild commented:
Pull request #18971 was updated. @ghellwig, @ianna, @kpedro88, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @perrotta, @civanch, @monttj, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @Dr15Jones, @kkousour, @mdhildreth, @cerminar, @slava77, @ggovi, @mulhearn, @arunhep, @dmitrijus, @franzoni can you please check and sign again.

mtosi commented:
I think you need a rebase

On Tue, Jun 13, 2017 at 11:08 AM, cmsbuild <notifications@github.com> wrote:

> Pull request #18971 <https://github.com/cms-sw/cmssw/pull/18971> was
> updated. @ghellwig <https://github.com/ghellwig>, @ianna
> <https://github.com/ianna>, @kpedro88 <https://github.com/kpedro88>,
> @fabozzi <https://github.com/fabozzi>, @Martin-Grunewald
> <https://github.com/martin-grunewald>, @rekovic
> <https://github.com/rekovic>, @vanbesien <https://github.com/vanbesien>,
> @perrotta <https://github.com/perrotta>, @civanch
> <https://github.com/civanch>, @monttj <https://github.com/monttj>,
> @silviodonato <https://github.com/silviodonato>, @cmsbuild
> <https://github.com/cmsbuild>, @fwyzard <https://github.com/fwyzard>,
> @GurpreetSinghChahal <https://github.com/gurpreetsinghchahal>,
> @davidlange6 <https://github.com/davidlange6>, @Dr15Jones
> <https://github.com/dr15jones>, @kkousour <https://github.com/kkousour>,
> @mdhildreth <https://github.com/mdhildreth>, @cerminar
> <https://github.com/cerminar>, @slava77 <https://github.com/slava77>,
> @ggovi <https://github.com/ggovi>, @mulhearn <https://github.com/mulhearn>,
> @arunhep <https://github.com/arunhep>, @dmitrijus
> <https://github.com/dmitrijus>, @franzoni <https://github.com/franzoni>
> can you please check and sign again.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18971#issuecomment-308055011>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58_OrMZrUja943bQj73KXgJSowa8-ks5sDlGXgaJpZM4Nn66Y>
> .
>

gomber committed:
Implemented D.Lange comments except the unique ptrs

cmsbuild commented:
Pull request #18971 was updated. @ghellwig, @ianna, @kpedro88, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @perrotta, @civanch, @monttj, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @Dr15Jones, @kkousour, @mdhildreth, @cerminar, @slava77, @ggovi, @mulhearn, @arunhep, @dmitrijus, @franzoni can you please check and sign again.

cmsbuild commented:
Pull request #18971 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

bhawnasinp commented:
@davidlange6 you wanted me to use unique_ptr for  GenericTriggerEventFlag* num_genTriggerEventFlag_; I tried using it following way 
std::unique_ptr<GenericTriggerEventFlag> num_genTriggerEventFlag_ (new GenericTriggerEventFlag);  
It gives me error. Do you have any suggestion for the same?

davidlange6 commented:
i don't follow your syntax I guess - you declare num_genTriggerEventFlag_  as a class data member, so you have to change its type in the header, and then change the syntax,similar to below, to set it in the cc. Is that what you did?

> On Jun 13, 2017, at 1:08 PM, bhawnasinp <notifications@github.com> wrote:
> 
> @davidlange6 you wanted me to use unique_ptr for GenericTriggerEventFlag* num_genTriggerEventFlag_; I tried using it following way
> std::unique_ptr num_genTriggerEventFlag_ (new GenericTriggerEventFlag);
> It gives me error. Do you have any suggestion for the same?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

gomber committed:
All comments from D.Lange implemeted

cmsbuild commented:
Pull request #18971 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
@davidlange6 so i fixed the issue, and everything is up-to date now from my side. Please have a look!

Dr15Jones commented on DQMOffline/Trigger/plugins/PhotonMonitor.cc:
```diff
@@ -54,8 +54,8 @@ PhotonMonitor::PhotonMonitor( const edm::ParameterSet& iConfig ) :
 {
-  if (num_genTriggerEventFlag_) delete num_genTriggerEventFlag_;
-  if (den_genTriggerEventFlag_) delete den_genTriggerEventFlag_;
+  if (num_genTriggerEventFlag_) num_genTriggerEventFlag_.release();
```
This leaks memory. `release()` returns the pointer and stops the ownership`. Instead, just remove lines 57 and 58. In fact, you can do away with the entire destructor since the compiler will do the right thing.

gomber replied:
ok thanks, I will remove the destructor and push it again.

gomber committed:
Removed .release() from the destructor

cmsbuild commented:
Pull request #18971 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
-1

Tested at: 651a9c966f088589c9b10bb20ed00b84272c8f5f


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/bd4d843bf0b01a6e0d376388e0f5b20f46203f02
https://github.com/cms-sw/cmssw/commit/935933def33dd70ac45836095e645eae823f2aa6
https://github.com/cms-sw/cmssw/commit/be1149239d5c6f32bdc59947f9cfbfe08ed9dd75
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20543/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20543/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20543/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/bd4d843bf0b01a6e0d376388e0f5b20f46203f02
https://github.com/cms-sw/cmssw/commit/935933def33dd70ac45836095e645eae823f2aa6
https://github.com/cms-sw/cmssw/commit/be1149239d5c6f32bdc59947f9cfbfe08ed9dd75
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20543/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20543/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20543/summary.html

Comparison Summary:
 - You potentially added 7 lines to the logs
 - Reco comparison results: 41 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1817858
 - DQMHistoTests: Total failures: 20168
 - DQMHistoTests: Total nulls: 130
 - DQMHistoTests: Total successes: 1797394
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

gomber commented:
@cmsbuild Looking at 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20543/unitTests.log
I see, the error is due to
"Input file file:dqm_missing.root appears to be corrupted since it does not have the proper internal structure. Check to see if the file was closed properly."

So I am not sure what can I do for the same?

davidlange6 commented:
the unit test error should not be related to this PR

davidlange6 commented:
hi @gomber - three additional comments (all repeated on N PRs now...)

1) add the package name to the include guard - eg, DQMOFFLINE_TRIGGER_PHOTON_H
2) make nbins an unsigned int everywhere
3) how many new histogram bins does this PR add?

gomber committed:
Implemented D.Lange comments about nbins and module

cmsbuild commented:
Pull request #18971 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
@davidlange6  I have implemeted your comments about package name and nbins as using int everywhere. And regarding your 3) point, this PR add 8 new histograms.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: 136562f76515f1cfc219c9f9f22a5dc6dc1d9509
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20609/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/20609/summary.html

Comparison Summary:
 - You potentially added 18 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1803170
 - DQMHistoTests: Total failures: 29476
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1773528
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

davidlange6 commented:
hi @gomber - its bins not histograms that we need to check (but to clarify  you have more than 8 histograms.. you have 8 MEs with multiple histograms and ~10 instances of your module. so maybe 160 histograms)

gomber commented:
Thanks @davidlange6! you are right,s its 160 histograms. Is there somethign more needs to be done or can we merge this PR?

davidlange6 commented:
how many _bins_?

gomber committed:
small fix

cmsbuild commented:
Pull request #18971 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

gomber committed:
Merge branch 'master' into EXOTrigger_promptMonitoring92X

cmsbuild commented:
Pull request #18971 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

gomber committed:
Merged EXOTrigger_promptMonitoring92X from repository gomber

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: c6ddc430562c067aa9ebb7a28ba601e3b3cd144a


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5387f33035f82196d51968058b52bc670f79533c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21060/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21060/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21060/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>gmake[1]: Target 'PostBuild' not remade because of errors.
gmake[1]: Leaving directory '/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000'
config/SCRAM/GMake/Makefile.rules:2058: recipe for target 'src' failed
gmake: *** [src] Error 2
gmake: Target 'all' not remade because of errors.
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5387f33035f82196d51968058b52bc670f79533c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21060/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21060/git-merge-result

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

gomber committed:
fixed small ug

cmsbuild commented:
Pull request #18971 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
@dmitrijus please test again, there was one small bug in the photon_cff file, which i fixed, looks like some typo happened in the last round.

gomber commented:
@fwyzard here it is https://github.com/cms-sw/cmssw/pull/19506

gomber commented:
@davidlange6 Regarding the number of bins, we have 16 histograms which corresponds to 2900 bins in total and then there are 10 instances of it, so that would mean in total this code would add 29000 bins, if this is the kind of information you asked me.

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 842acda232ac7e704446a0d04dcde7136ea91c81
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21073/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b1580cdb9b247f8d4a93ce44ec73c8c8213a595e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21073/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21073/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18971/21073/summary.html

Comparison Summary:
 - You potentially added 10 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1756063
 - DQMHistoTests: Total failures: 102
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1755795
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

gomber commented:
@davidlange6 Could you please proceed with this? as all the tests are passed and we also provide nbins information?

davidlange6 commented:
all that happened a few hours ago:)

davidlange6 commented:
merge
